README.md

```diff
@@ -4,6 +4,103 @@
 [![GitHub release](https://img.shields.io/github/release/jenkinsci/kubernetes-credentials-provider-plugin.svg?label=release)](https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/releases/latest)
 [![Jenkins Plugin Installs](https://img.shields.io/jenkins/plugin/i/kubernetes-credentials-provider.svg?color=blue)](https://plugins.jenkins.io/kubernetes-credentials-provider)
 
-Jenkins credentials provider for Kubernetes
+Complete documentation on GitHub page is [here](https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/)
 
-Documentation is [here](https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/)
+The *Kubernetes Credentials Provider* is a [Jenkins](https://jenkins.io) plugin to enable the retreival of [Credentials](https://plugins.jenkins.io/credentials) directly from Kubernetes.
+
+The plugin supports most common credential types and defines an [`extension point`](https://jenkins.io/doc/developer/extensions/kubernetes-credentials-provider/) that can be implemented by other plugins to add support for custom Credential types. 
```

Comment:
Used the GitHub page for this one instead of a relative path as it contains macros, not rendered on GitHub.